doc/user/content/sql/types/list.md

```diff
@@ -0,0 +1,490 @@
+---
+title: "List Data Types"
+description: "Lists are an ordered, multidimensional sequence of any type, including other lists"
+menu:
+  main:
+    parent: sql-types
+---
+
+{{< version-added v0.5.1 >}}
+
+Lists are an ordered, multidimensional sequence of any type, including other lists.
```

Comment:
I worry that using the term "multidimensional" to refer to a nested list may be technically incorrect because a "dimension" implies a uniform length (i.e., what arrays have but lists do not). I could be crazy, so maybe @umanwizard can weigh in too, since he comes from a world that uses this terminology precisely (PyTorch).

Reply:
In PyTorch (and probably Numpy though idk), indeed, arrays can't be jagged -- but I think that's an implementation detail, rather than having arisen from the precise meaning of technology.

There is by now though a lot of inertia behind terms like "Ndarrays" and "tensors" to refer to specific things, so maybe it'd be best if we avoid them (and that would include avoiding the term "dimension", probably). I have no strong opinion.

Reply:
This may be clear to devs (maybe it's what "multidimensional" implies?), but I can't tell from the definition if this can only be a sequence of multiple items all of the same type, or can also include multiple items of different types.

Also, this page doesn't show up for me anywhere in the left navigation -- I can only go to it by going directly to sql/types/list.

Reply:
Chiming in to say I don't think it's any clearer to devs whether lists must be homogeneous or not.